doc/user/CONTRIBUTING.md

```diff
@@ -0,0 +1,55 @@
+This doc is a WIP. There is more useful information in the [README](./README.md).
+
+# Set up
+
+Rather than work on `github.com/materializeinc/materialize` itself, we recommend
+working on your own fork of Materialize; this makes branches much tidier.
+branches tidier.
```

Comment:
Potential nitpick: only people with write access to this repo can create a branch inside it, all external contributors are forced to fork it.
I guess this depends on the target audience of this document: people at Materialize or contributors in general